src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs

```diff
@@ -383,7 +399,7 @@ public async ValueTask<Stream> GetStreamAsync(BalancerAddress address, Cancellat
                 SocketConnectivitySubchannelTransportLog.ClosingSocketFromIdleTimeoutOnCreateStream(_logger, _subchannel.Id, address, _socketIdleTimeout);
                 closeSocket = true;
             }
-            else if (IsSocketInBadState(socket, address))
+            else if (CheckSocket(socket, address, ref socketData, out _))
```

Comment:
Is this the only caller that needs to be updated?  When would you want to make a raw `IsSocketInBadState` call?

Reply:
You wouldn't. It now is only used by the method that also consumes data. I'll leave it as its own method to prevent the other one from getting too big. And add a comment.